src/js/webrequest.js

```diff
@@ -738,6 +738,31 @@ function dispatcher(request, sender, sendResponse) {
     badger.storage.revertUserAction(request.origin);
     sendResponse();
 
+  } else if (request.type == "downloadWhitelist") {
+    chrome.storage.sync.get("disabledSites", function (store) {
+      if (store.hasOwnProperty("disabledSites")) {
+        var settings = badger.getSettings();
+        var whitelist = _.union(settings.getItem("disabledSites"), store.disabledSites);
+        settings.setItem("disabledSites", whitelist);
+        sendResponse(true);
+      } else {
+        sendResponse(false);
+      }
+    });
+    return true; //Let internal asychronus call continue
```

Comment:
typo: "asynchronous"

Reply:
Should we reuse previously-used comment text, "indicate this is an async response to chrome.runtime.onMessage"?

Reply:
I had not noticed that comment before, I have changed them.